v0/destinations/mp/deleteUsers.js

```diff
@@ -0,0 +1,59 @@
+const { httpPOST } = require("../../../adapters/network");
+const ErrorBuilder = require("../../util/error");
+
+/**
+ * This function will help to delete the users one by one from the userAttributes array.
+ * @param {*} userAttributes Array of objects with userId, emaail and phone
+ * @param {*} config Destination.Config provided in dashboard
+ * @returns
+ */
+const userDeletionHandler = async (userAttributes, config) => {
+  if (!Array.isArray(userAttributes)) {
+    throw new ErrorBuilder()
+      .setMessage("userAttributes is not an array")
+      .setStatus(400)
+      .build();
+  }
+  if (!config?.token) {
+    throw new ErrorBuilder()
+      .setMessage("API Token is not available to make")
+      .setStatus(400)
+      .build();
+  }
+  const params = {
+    data: {
+      $token: `${config.token}`,
+      $delete: null,
+      $ignore_alias: true
+    }
+  };
+  const endpoint =
+    config.dataResidency === "eu"
+      ? `https://api-eu.mixpanel.com/engage`
+      : `https://api.mixpanel.com/engage`;
+  for (let i = 0; i < userAttributes.length; i += 1) {
```

Comment:
lets use `for of` here

Reply:
If we use `for of` it is showing some lint error. If we use for of it will create a different iterator at each step, whereas if we use normal `for` it will access the same memory space in every step.